sklearn/impute/tests/test_impute.py

```diff
@@ -445,6 +449,22 @@ def test_imputation_constant_pandas(dtype):
     assert_array_equal(X_trans, X_true)
 
 
+def test_simple_imputation_splitting_missing_value_in_test_array():
+    # [Non Regression Test] Missing value in test set should not throw an error
+    # and return a finite dataset
+    X = np.array([1, 2, 3, np.nan]).T
+    y = np.array([0, 1, 1, 1])
+    test_fold = np.array([0, 1, 0, 1])
+
+    ps = PredefinedSplit(test_fold)
+    pipe1 = make_pipeline(SimpleImputer(add_indicator=True),
+                          LogisticRegression(solver='lbfgs'))
+    scores = cross_val_score(pipe1, X, y, cv=ps)
```

Comment:
I think it would be clearer to not use these tools, and just do `imputer.fit([[1], [2]]).transform([[3], [nan]])`. In master, I think, this will fail. In this PR it should not.

Reply:
Certainly your test is sufficient...  just excessive machinery and hence unclear what it is testing.

Reply:
Thanks for the input! I spent a lot of time trying to understand how the PredefinedSplit in order to write the test...turns out there was an easier approach.